tests/manage/z_cluster/nodes/test_node_replacement_reactive_aws_ipi.py

```diff
@@ -0,0 +1,146 @@
+import logging
+import pytest
+import random
+from ocs_ci.framework.testlib import (
+    tier4, tier4b, ManageTest, aws_platform_required,
+    ipi_deployment_required, ignore_leftovers)
+from ocs_ci.ocs import machine
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources import pod
+from tests.helpers import wait_for_resource_state
+from tests.sanity_helpers import Sanity
+from tests.helpers import get_worker_nodes
+from ocs_ci.ocs import ocp
+
+log = logging.getLogger(__name__)
+
+
+@ignore_leftovers
+@tier4
+@tier4b
+@aws_platform_required
+@ipi_deployment_required
+class TestNodeReplacement(ManageTest):
+    """
+    Knip-894 Node replacement - AWS-IPI-Reactive
+    """
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+
+        """
+        self.sanity_helpers = Sanity()
+
+    def test_node_replacement_reactive_aws_ipi(
+            self, nodes, pvc_factory, pod_factory,
+            dc_pod_factory
+    ):
+        """
+        Knip-894 Node replacement - AWS-IPI-Reactive
+
+        """
+        # Get worker nodes
+        initial_nodes = get_worker_nodes()
+        log.info(f"Current available worker nodes are {initial_nodes}")
+
+        # Get the osd associated node
+        osd_pods_obj = pod.get_osd_pods()
+        osd_node_obj = pod.get_pod_node(random.choice(osd_pods_obj))
+        log.info(f"Selected OSD is {osd_node_obj.name}")
+
+        # Create fedora dc app on all the worker nodes and start IO in
+        # background
+        dc_rbd_pod_obj = []
+        for node in initial_nodes:
+            # Create app pods on all the nodes
+            dc_rbd = dc_pod_factory(
+                interface=constants.CEPHBLOCKPOOL, node_name=node)
+            if node == osd_node_obj.name:
+                pod.run_io_in_bg(dc_rbd, expect_to_fail=True, fedora_dc=True)
+            else:
+                pod.run_io_in_bg(dc_rbd, expect_to_fail=False, fedora_dc=True)
+            dc_rbd_pod_obj.append(dc_rbd)
+
+        dc_cephfs_pod_obj = []
+        for node in initial_nodes:
+            # Create app pods on all the nodes
+            dc_cephfs = dc_pod_factory(
+                interface=constants.CEPHFILESYSTEM, node_name=node)
+            if node == osd_node_obj.name:
```

Comment:
please set expect_to_fail prior to calling run_io_in_bg() and then pass this var to the function so it will be called only once

Reply:
Done